backend/kesaseteli/applications/views.py

```diff
@@ -163,3 +170,123 @@ def export_and_download_annual_applications(
+
+    @staticmethod
+    def source_fields_and_output_names() -> List[Tuple[str, str]]:
+        with translation.override("fi"):
+            return [
+                ("application_date", str(_("Hakupvm"))),
+                ("id", str(_("Tunniste"))),
+                ("status", str(_("Hakemuksen tila"))),
+                ("application_year", str(_("Hakuvuosi"))),
+                ("summer_voucher_serial_number", str(_("Kesäsetelinro"))),
+                ("birth_year", str(_("Syntymävuosi"))),
+                ("birthdate", str(_("Syntymäpvm"))),
+                ("first_name", str(_("Etunimi"))),
+                ("last_name", str(_("Sukunimi"))),
+                ("vtj_last_name", str(_("VTJ-sukunimi"))),
+                ("school", str(_("Koulu"))),
+                ("is_unlisted_school", str(_("Listaamaton koulu?"))),
+                ("email", str(_("Sähköposti"))),
+                ("phone_number", str(_("Puhelinnro"))),
+                ("postcode", str(_("Postinro"))),
+                ("vtj_home_municipality", str(_("VTJ-kotikunta"))),
+                ("additional_info_user_reasons", str(_("Lisätietosyyt"))),
+                ("additional_info_description", str(_("Lisätiedot"))),
+                ("language", str(_("Kieli"))),
+                ("confirmation_date", str(_("Vahvistettu"))),
+                ("additional_info_providing_date", str(_("Lisätiedot annettu"))),
+                ("handling_date", str(_("Käsitelty"))),
+            ]
+
+    @staticmethod
+    def source_fields() -> Tuple[str, ...]:
+        return list(
+            zip(*YouthApplicationExcelExportView.source_fields_and_output_names())
+        )[0]
+
+    @staticmethod
+    def output_column_names() -> Tuple[str, ...]:
+        return list(
+            zip(*YouthApplicationExcelExportView.source_fields_and_output_names())
+        )[1]
+
+    def get_queryset(self):
+        return YouthApplication.objects.active().order_by("created_at", "pk")
+
+    def serializer(self, apps: QuerySet[YouthApplication]):
+        return (self.generate_data_row(app) for app in apps)
+
+    @enforce_handler_view_adfs_login
+    def get(
+        self, request, *args, **kwargs
+    ) -> Union[HttpResponse, HttpResponseRedirect, StreamingHttpResponse]:
+        queryset = self.get_queryset()
+        if not queryset.exists():
+            return HttpResponse(_("Hakemuksia ei löytynyt."))
+
+        response = StreamingHttpResponse(
+            xlsx_streaming.stream_queryset_as_xlsx(
+                qs=queryset,
+                xlsx_template=self.xlsx_template(queryset),
+                serializer=self.serializer,
+                batch_size=settings.EXCEL_DOWNLOAD_BATCH_SIZE,
+            ),
+            content_type="application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+        )
+        response["Content-Disposition"] = f"attachment; filename={self.xlsx_filename}"
+        return response
+
+    @property
```

Comment:
Maybe this unnecessary nagging, but I would probably move the excel-related functions to another excel-utility class because of S in [SOLID principles](https://en.wikipedia.org/wiki/SOLID) eg. "every class should have only one responsibility"

Reply:
I'd rather not refactor the Excel exporting code in this pull request as there is a lot of overlap with the employer applications' Excel export in the same file. I knew when making this code that there's going to be overlap and code duplication but it was clearer and less time costly to do it this way now rather than try to refactor the two Excel export functionalities together in a clean way—I did try to combine them together first but that implementation became convoluted and I decided against using it. So yes, this code could be improved upon but I'd rather not go into it in this pull request now to save time as there's a lot that could be combined between `YouthApplicationExcelExportView` and `EmployerApplicationExcelDownloadView`.